app/src/main/java/com/android/unio/MainActivity.kt

```diff
@@ -116,10 +118,18 @@ fun UnioApp(authViewModel: AuthViewModel) {
     }
     navigation(startDestination = Screen.HOME, route = Route.HOME) {
       composable(Screen.HOME) {
-        HomeScreen(
-            navigationActions,
-            eventListViewModel = eventListViewModel,
-            userViewModel = userViewModel)
+        HomeScreen(navigationActions, eventListViewModel, userViewModel = userViewModel)
+      }
+      composable(Screen.EVENT_DETAILS) { navBackStackEntry ->
+        // Get the event UID from the arguments
+        val uid = navBackStackEntry.arguments?.getString("uid")
+        // Create the Event screen with the event UID
+        uid?.let {
+          EventScreen(
+              navigationAction = navigationActions,
+              eventViewModel = EventViewModel(eventRepository, userRepository),
```

Comment:
You should extract the instanciation of the `EventViewModel` at the top of the composable, and wrap it in a `remember { ... }`